[45기 연정아] Assignment 2 | Mock Data를 활용하여 여러 개의 피드 구현 (리팩토링 완료, 수정 완료)

jeonga-yeon opened the pull request:
수고하셨습니다.

- 이 브랜치에서 개발한 내용을 정리하는 PR 본문에 어떤걸 적어야할지 간략한 가이드를 준비했습니다.
- **(예시)** 라고 되어있는 부분은 정말 <u>예시</u> 입니다.
- **(예시)** 코드는 지우고, 여러분의 개발 기록을 남겨주세요.

---

##### 0. map 함수 적용시 key prop을 부여하는 이유
 불필요한 리렌더링을 막기 위해서 엘리먼트에 고유성을 부여하고자 key props를 부여합니다.  
 예를 들어, 여러 개의 리스트가 있을 때 그 중 하나만 변경 되어도 전체가 리렌더링 되는데, 그런 경우를 방지할 수 있습니다.
 key props를 부여함으로써 React가 어떤 항목을 변경, 추가, 삭제할지 식별하도록 도울 수 있습니다.

##### 1. 이 브랜치에서 어떤걸 개발했는지 title,그리고 detail을 적어주세요.

##### [Login] 사용자 입력 데이터 저장

- ID <input> 에서 onChange event가 발생
- event 발생 시 saveUserId 함수가 실행
- saveUserId 함수는 이벤트를 인자로 받음
- event가 일어난 요소에 담긴 value 값(event.target.value)을 state에 저장
- Password <input> 에도 동일하게 적용

##### [Login] 로그인 버튼 활성화(validation)

- ID에 '@'이 포함되고 Password가 5글자 이상인 경우 로그인 버튼 색상이 활성화 되고 disabled 여부가 조작됨

##### [Main] 댓글 기능

- 사용자가 댓글 입력 후 enter 키를 누르거나 댓글 입력 버튼 클릭 시 댓글이 추가되도록 구현
- 댓글을 컴포넌트화 하고 props로 데이터 전달


<br />

##### 2. 개발한 화면을 캡쳐해서 첨부 해 주세요. (drag & drop 또는 첨부파일 추가)

<br />

##### 3. 이 브랜치에서 개발하면서 느꼇던 성장포인트를 적어주세요.

- 조건부 렌더링을 할 수 있게 됐습니다.
- 부모 컴포넌트로부터 자식 컴포넌트로 props를 보내 데이터를 활용할 수 있게 되었습니다.
- map 함수를 사용해 컴포넌트를 생성할 수 있게 되었습니다.

  <br />

jeonga-yeon committed:
ADD: Main과 Login 기능 추가

jeonga-yeon committed:
Add : Mission 1 - 사용자 입력 데이터 저장 기능 구현

jeonga-yeon committed:
Add : Mission 2 - 로그인 버튼 활성화 기능 구현

jeonga-yeon committed:
Add : Mission 3 - 댓글 기능 구현

jeonga-yeon committed:
Add : Mission 4 - 댓글 컴포넌트화, Props로 데이터 전달 구현

jeonga-yeon committed:
Modify: 댓글 입력창이 공백일 경우 댓글 입력 못하게 수정

jeonga-yeon committed:
fix: 오타 수정

sstaar91 commented on src/pages/jeongaYeon/Login/Login.js:
```diff
@@ -1,8 +1,76 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Login.scss';
+import { useNavigate } from 'react-router-dom';
```
import 순서 수정해주세요! 일반적인 convention을 따라 순서만 잘 지켜주셔도 가독성이 좋아집니다. 
아래 순서 참고해주세요.

React → Library(Package) → Component → 변수 / 이미지 → css 파일(scss 파일)

sstaar91 commented on src/pages/jeongaYeon/Login/Login.js:
```diff
@@ -1,8 +1,76 @@
+  };
+
+  const saveUserId = event => {
+    let userId = event.target.value;
+    setId(userId);
```
```suggestion
    setId(event.target.value);
```
따로 변수로 뺄 필요없이 이렇게 바로 넣어줘도 되겠네요!

sstaar91 commented on src/pages/jeongaYeon/Login/Login.js:
```diff
@@ -1,8 +1,76 @@
+  };
+
+  const saveUserPassword = event => {
+    let userPassword = event.target.value;
+    setPassword(userPassword);
```
여기도 위의 리뷰와 마찬가지입니다~

sstaar91 commented on src/pages/jeongaYeon/Login/Login.js:
```diff
@@ -1,8 +1,76 @@
+  // }, [password, id]);
+
+  return (
+    <main className="container">
```
sass 사용할 때, 가장 최상위 부모태그의 이름은 컴포넌트의 이름과 동일하게 해주는 것이 좋다고 했습니다!
해당 부분에 맞게 수정해주세요~

sstaar91 commented on src/pages/jeongaYeon/Login/Login.js:
```diff
@@ -1,8 +1,76 @@
+            <button
+              className={valid ? 'loginButton loginButtonBlue' : 'loginButton'}
+              onClick={handleLogin}
+              disabled={valid ? false : true}
```
valid 라는 값이 결국 true or false의 boolean 값이기 때문에
삼항연산자로 true, false를 적용하는것은 불필요한 코드입니다!
수정 해주세요~

sstaar91 commented on src/pages/jeongaYeon/Login/Login.scss:
```diff
@@ -0,0 +1,64 @@
+  .loginButton {
+    width: 100%;
+    height: 35px;
+    margin-top: 20px;
+    margin-bottom: 180px;
```
```suggestion
    margin: 20px 0 180px ;
```
가능하다면 코드를 짧게 줄이는 것이 좋습니다!
위의 두 스타일은 이렇게 합칠 수 있겠습니다.

sstaar91 commented on src/pages/jeongaYeon/Main/Main.scss:
```diff
@@ -0,0 +1,365 @@
+        align-items: center;
+        justify-content: space-between;
+
+        span:first-child {
+          color: gray;
+        }
+
+        span:last-child {
+          font-weight: bold;
+        }
```
header 에 포함된 부분이라면 & 연산자를 사용해주세요

jeonga-yeon replied:
& 연산자를 어디에 적용해야 할지 잘 모르겠습니다!

sstaar91 commented on src/pages/jeongaYeon/Main/Main.scss:
```diff
@@ -0,0 +1,365 @@
+        align-items: center;
+        margin: 15px 0;
+
+        img {
+          width: 40px;
+          height: 40px;
+          margin-right: 10px;
+          border-radius: 50%;
+          object-fit: cover;
+        }
```
태그에 직접 스타일을 적용하는 것은
자칫 다른 컴파일 과정에서 다른 동일한 태그의 스타일에도 영향을 줄 수 있습니다!
스타일을 부여할 때는 className을 이용해서 부여하는 방향으로 구현해주세요!

sstaar91 reviewed (commented):
정아님 수고하셨습니다!
수정 사항 반영해주세요~

jeonga-yeon committed:
Modify: 로그인 유효성 검사에 useEffect 추가 & Main 화면 CSS 수정

jeonga-yeon committed:
Modify: Main과 Login 수정
- 컨벤션에 따라 import 순서 수정
- 불필요한 변수 삭제
- css 클래스 네임 추가
- css 수정

jeonga-yeon committed:
Modify: Login 버튼 조건부 렌더링 수정

jeonga-yeon committed:
Assignment 1 | Aside에서 상수 데이터 활용

Jinny-Jin commented on src/pages/jeongaYeon/Login/Login.js:
```diff
@@ -1,8 +1,63 @@
+  const checkValid = id.indexOf('@') === -1 || password.length < 5;
+
+  return (
+    <main className="login">
```
정아님 체고...♥

Jinny-Jin reviewed (commented):
LGTM

jeonga-yeon committed:
Add: 데이터를 받아와서 피드를 구성하도록 수정

jeonga-yeon committed:
Assignment 2 | Mock Data를 활용하여 여러 개의 피드 구현

reveur96 commented on src/pages/jeongaYeon/Main/Main.js:
```diff
@@ -9,6 +9,7 @@ const MainJeonga = () => {
 
   const handleComment = event => {
     const comment = event.target.value;
+    if (comment.length <= 0) return;
```
value값이 없음을 length로 비교하신 이유가 있나요? 궁금합니다^^~*

jeonga-yeon replied:
value가 string이라 length가 0이면 값이 없는 거라고 생각해서 넣었어요ㅎㅎ

reveur96 reviewed (commented):
좋은 코드 잘보고 갑니다^^

reveur96 commented on public/data/feed.json:
```diff
@@ -0,0 +1,109 @@
+        {
+          "id": 1,
+          "commentUser": "jelly",
+          "commentContent": "해삐해삐"
```
☆*: .｡. o(≧▽≦)o .｡.:*☆

reveur96 commented on src/pages/jeongaYeon/Login/Login.js:
```diff
@@ -1,8 +1,63 @@
+    setPassword(event.target.value);
+  };
+
+  const checkValid = id.indexOf('@') === -1 || password.length < 5;
```
여기서 id값을 -1과 비교하는 이유가 있나요?(●'◡'●)

jeonga-yeon replied:
indexOf 함수를 사용하면 id에 @가 없을 때 index 값이 -1로 나오기 때문에 id에 @가 있는지 없는지 확인하기 위해 비교했어요!(●'◡'●)

Jinny-Jin commented on public/data/feed.json:
```diff
@@ -0,0 +1,109 @@
+        }
+      ]
+    }
+  ]
```
호,,고생 많으셨어유 🤩

Jinny-Jin commented on src/pages/jeongaYeon/Main/Feed.js:
```diff
@@ -0,0 +1,110 @@
+  const [comments, setComments] = useState([]);
+
+  const handleComment = event => {
+    const comment = event.target.value;
+    if (comment.length <= 0) return;
+    setValue(comment);
+    if (event.keyCode === 13) {
+      setComments(current => [...current, comment]);
+      setValue('');
+    }
+  };
```
저는 생각도 못했던,, 👍

jeonga-yeon committed:
Refactor: westagram 리팩토링
- 컨벤션에 따라 import 순서 변경
- 계산된 속성명을 이용한 input Handler 함수 합치기
- variables.scss에 변수 추가

sstaar91 commented on src/pages/jeongaYeon/Main/Feed.js:
```diff
@@ -0,0 +1,110 @@
+  };
+
+  return (
+    <article className="feedsWrapper">
```
```suggestion
    <article className="feed">
```
컴포넌트의 이름과 동일하게 하려면 이렇게 하는 게 맞겠네요!

jeonga-yeon replied:
수정 완료했습니다!!

sstaar91 reviewed (commented):
정아님 수고많으셨습니다!

Jinny-Jin commented on public/data/feed.json:
```diff
@@ -8,7 +8,14 @@
       "heartUser": "youworld",
       "heartUsers": "10",
-      "feedContent": "너-무 행복했던 4월 1일💕"
+      "feedContent": "너-무 행복했던 4월 1일💕",
```
넘행사 😎

jeonga-yeon committed:
Modify: class명 컴포넌트 이름과 동일하게 수정
